core/src/main/java/tc/oc/pgm/modules/StatsMatchModule.java

```diff
@@ -0,0 +1,299 @@
+    Component killstreakMessage = getKillstreakMessage(sortStats(allKillstreaks), match);
+    Component deathMessage = getDeathsMessage(sortStats(allDeaths), match);
+    TopResult bestBowshot = sortStats(allBowshots);
+    Component bowshotMessage = getBowshotMessage(bestBowshot, match);
+
+    for (MatchPlayer viewer : match.getPlayers()) {
+      viewer.sendMessage(
+          Components.fromLegacyText(
+              ComponentUtils.horizontalLineHeading(
+                  ChatColor.YELLOW
+                      + AllTranslations.get().translate("stats.best", viewer.getBukkit()),
+                  ChatColor.WHITE,
+                  ComponentUtils.MAX_CHAT_WIDTH)));
+
+      viewer.sendMessage(killMessage);
+      viewer.sendMessage(killstreakMessage);
+      viewer.sendMessage(deathMessage);
+      if (bestBowshot.stat != 0)
+        viewer.sendMessage(bowshotMessage); // Prevent from showing if bows are not used
+    }
+  }
+
+  private static class TopResult {
+    UUID uuid;
+    int stat;
+  }
+
+  private TopResult sortStats(Map<UUID, Integer> map) {
+    Map.Entry<UUID, Integer> mapEntry =
+        map.entrySet().stream().max(Comparator.comparingInt(Map.Entry::getValue)).orElse(null);
+    TopResult topResult = new TopResult();
+
+    if (mapEntry == null) { // Should never happen, but acts as a failsafe
+      topResult.uuid = UUID.fromString("3c7db14d-ac4b-4e35-b2c6-3b2237f382be");
+      topResult.stat = 0;
+      return topResult;
+    }
+    topResult.uuid = mapEntry.getKey();
+    topResult.stat = mapEntry.getValue();
+    return topResult;
+  }
+
+  Component getKillsMessage(TopResult topResult, Match match) {
+    return new Component(
+        new PersonalizedTranslatable(
+                "stats.kills",
+                playerName(match, topResult.uuid),
+                new PersonalizedText(Integer.toString(topResult.stat), ChatColor.GREEN))
+            .render());
+  }
+
+  Component getKillstreakMessage(TopResult topResult, Match match) {
+    return new Component(
+        new PersonalizedTranslatable(
+                "stats.killstreak",
+                playerName(match, topResult.uuid),
+                new PersonalizedText(Integer.toString(topResult.stat), ChatColor.GREEN))
+            .render());
+  }
+
+  Component getDeathsMessage(TopResult topResult, Match match) {
+    return new Component(
+        new PersonalizedTranslatable(
+                "stats.deaths",
+                playerName(match, topResult.uuid),
+                new PersonalizedText(Integer.toString(topResult.stat), ChatColor.RED))
+            .render());
+  }
+
+  Component getBowshotMessage(TopResult topResult, Match match) {
+    return new Component(
+        new PersonalizedTranslatable(
+                "stats.bowshot",
+                playerName(match, topResult.uuid),
+                new PersonalizedText(Integer.toString(topResult.stat), ChatColor.YELLOW))
+            .render());
+  }
```

Comment:
These methods are actually them same. The only difference seems to be translation keys and colors. To keep this try could try make a generic method with these as parameters, and these methods would call it.

```java
Component getBowshotMessage(TopResult topResult, Match match) {
  return getMessage("stats.bowshot", topResult, match, ChatColor.YELLOW);
}

Component getMessage(String messageKey, TopResult topResult, Match match, CharColor color) {
  return new Component(
      new PersonalizedTranslatable(
              messageKey,
              playerName(match, topResult.uuid),
              new PersonalizedText(Integer.toString(topResult.stat), color))
          .render());
}
```

Reply:
Do we even need a Component constructor for each message then? Why not just call `getMessage` from a method directly.